klippy/extras/garbage_collection.py

```diff
@@ -0,0 +1,26 @@
+import gc
+import logging
+
+class GarbageCollection:
+    def __init__(self, config):
+        self.printer = config.get_printer()
+        can_freeze = hasattr(gc, 'freeze') and hasattr(gc, 'unfreeze')
+        if can_freeze and config.getboolean("freeze", True):
```

Comment:
I put a config option in there, so the freeze functionality _can_ be disabled if for whatever reason a user needs to disable it. I'm not sure if we want to document it in the config reference. If we do, any thoughts where to put it? I didn't see a section for configs that are more related to klippy functionality vs all the parts of the printer.